### ISSUE TITLE
Missing locale data for custom locale

### ISSUE THREAD

**Issue by DanielSanudo** [formatjs/react-intl#1483]
### Expected behavior

A custom locale is supported.

### Current behavior

When trying to use a custom locale we get the following error:
`[React Intl] Missing locale data for locale: "qa-PS". Using default locale: "en" as fallback.`

Before `react-intl@3` we were using the following code in order to support the custom locale:
```
import { addLocaleData } from 'react-intl';
addLocaleData({ locale: 'qa-PS' });
```

Now that `addLocaleData` was removed I can't find another way for adding a custom locale.

### Your Environment

|          Executable | Version |
| ------------------: | :------ |
| `npm ls react-intl` | 3.2.1 |
|      `npm ls react` | 16.9.0 |
|     `npm --version` | 6.4.1 |
|    `node --version` | 8.12.0 |

| OS   | Version |
| macOS Sierra | 10.14.5 |
| NAME | VERSION |

| Browser | Version |
| ------- | ------- |
| NAME    | VERSION |

**Comment by longlho** [comment]
Please take a look at our Upgrade Guide in docs

**Comment by DanielSanudo** [comment]
@longlho thanks for answering. I tried to look at the documentation and this is what I was able to find related to the issue.
https://github.com/formatjs/react-intl/blob/master/docs/Upgrade-Guide.md#migrate-to-using-native-intl-apis

I tried to use the polyfills and reproduce what this code was doing:
```
require('@formatjs/intl-relativetimeformat/dist/locale-data/de')
require('@formatjs/intl-pluralrules/dist/locale-data/de');
```
I ended up using something like this:
```
require('@formatjs/intl-pluralrules/polyfill');
Intl.PluralRules.__addLocaleData({ locale: 'qa-PS' });
require('@formatjs/intl-relativetimeformat/polyfill');
Intl.RelativeTimeFormat.__addLocaleData({ locale: 'qa-PS' });
```

This allowed me to display my page in the custom language somehow but I am still getting the message error.

Also I went deep into the code that triggers the message error and I ended up reaching this file ( I don't see how the code that I am trying can affect it ):
https://github.com/formatjs/formatjs/blob/master/packages/intl-locales-supported/src/index.ts

Am I missing something? is it possible to do what we are trying?

**Comment by longlho** [comment]
for `qa` u gotta modify the polyfill locale-data path to 
```
require('@formatjs/intl-pluralrules/dist/locale-data/qa');
require('@formatjs/intl-relativetimeformat/dist/locale-data/qa')
```

**Comment by longlho** [comment]
Looks like `qa-PS` is not a valid locale and `qa` is not a valid language. If it's Arabic it's `ar` so:
```
require('@formatjs/intl-pluralrules/dist/locale-data/ar');
require('@formatjs/intl-relativetimeformat/dist/locale-data/ar')
```

**Comment by DanielSanudo** [comment]
@longlho You are right, `qa` is not an existing locale. One of our teams have a custom / new locale for testing and before we were able to use it. Is not possible now?

**Comment by longlho** [comment]
TL;DR: Ignore the warning, everything should still work w/ custom locale https://codesandbox.io/s/heuristic-wildflower-gykb7

So the `addLocaleData` before was for `Intl.PluralRules` & `Intl.RelativeTimeFormat` which are now built into browsers:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/PluralRules
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RelativeTimeFormat

`react-intl` itself doesn't inherently have any locale-data.

**Comment by DanielSanudo** [comment]
@longlho Thank you for your help. I will be closing the issue.

**Comment by qrohlf** [comment]
As a note to anyone encountering this error, a side effect of the warning is that accessing `intl.locale` in a component using `injectIntl` will give you a locale of `en` instead of your custom locale.

I don't have a great workaround for this, yet.
